auth.go

```diff
@@ -53,7 +62,10 @@ func Authenticate(username, password, apiEndpoint string) (*Client, error) {
 		}
 
 		infolog.Printf("Authentication was successful")
-		cache.Set(getCacheKey(username, apiEndpoint), authResp.AccessToken, time.Duration(authResp.ExpiresIn)*time.Millisecond)
+		err = tokenCache.Put(getCacheKey(username, apiEndpoint), authResp.AccessToken, time.Duration(authResp.ExpiresIn)*time.Millisecond)
+		if err != nil {
+			return nil, ConstructNestedError("failed to cache the authentication token", err)
+		}
```

Comment:
We should definitely print debug info here, but should we error out if caching failed? I think the execution can still go ahead in this case.